pkg/hub/deployer/deployer.go

```diff
@@ -256,27 +356,84 @@ func (deployer *Deployer) getChartsBySelector(subs *appsapi.Subscription, feed a
 	return []*appsapi.HelmChart{}, nil
 }
 
-func (deployer *Deployer) populateDescriptionsForHelm(subs *appsapi.Subscription, chartRefs []appsapi.ChartReference) error {
+func (deployer *Deployer) getManifestsBySelector(subs *appsapi.Subscription, feed appsapi.Feed) ([]*appsapi.Manifest, error) {
+	namespace := subs.Namespace
+	if len(feed.Namespace) != 0 {
+		namespace = feed.Namespace
+	}
+	if len(feed.Name) > 0 {
+		manifest, err := deployer.manifestLister.Manifests(namespace).Get(feed.Name)
+		if err != nil {
+			return nil, err
+		}
+		return []*appsapi.Manifest{manifest}, nil
+	}
+
+	selector := labels.NewSelector()
+	if len(feed.Kind) > 0 {
+		requirement, err := labels.NewRequirement(known.ConfigKindLabel, selection.Equals, []string{feed.Kind})
+		if err != nil {
+			return nil, err
+		}
+
+		selector = selector.Add(*requirement)
+	}
+
+	if len(feed.APIVersion) > 0 {
+		requirement, err := labels.NewRequirement(known.ConfigVersionLabel, selection.Equals, []string{feed.APIVersion})
+		if err != nil {
+			return nil, err
+		}
+
+		selector = selector.Add(*requirement)
+	}
+
+	if feed.FeedSelector != nil {
+		feedSelector, err := metav1.LabelSelectorAsSelector(feed.FeedSelector)
+		if err != nil {
+			return nil, err
+		}
+
+		reqs, _ := feedSelector.Requirements()
+		for _, r := range reqs {
+			selector.Add(r)
+		}
+	}
+
+	if selector.Empty() {
+		return []*appsapi.Manifest{}, nil
+	}
+
+	manifestList, err := deployer.manifestLister.Manifests(namespace).List(selector)
+	if err != nil {
+		return nil, err
+	}
+	return manifestList, nil
+}
+
+func (deployer *Deployer) populateDescriptionsForHelm(subs *appsapi.Subscription, chartRefs []appsapi.ChartReference) (errs []error) {
 	var mcls []*clusterapi.ManagedCluster
 	for _, subscriber := range subs.Spec.Subscribers {
 		selector, err := metav1.LabelSelectorAsSelector(subscriber.ClusterAffinity)
 		if err != nil {
-			return err
+			errs = append(errs, err)
+			return
 		}
 		clusters, err := deployer.clusterLister.ManagedClusters("").List(selector)
 		if err != nil {
-			return err
+			errs = append(errs, err)
```

Comment:
ditto

Reply:
refector population for helm and solve these problems